src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
+    (m) => m.name === ch.mother
+  ).born);
+
+  return sum / children.length;
 }
```

Comment:
let's make this code more readable.
According to the condition of the task, we are only interested in moms, so we will take the search for mom into a separate function and then we can immediately get a list of what we are interested in.
After that, all we have to do is filter the array depending on the `onlyWithSons` variable and find the age between the child and the mother

```suggestion
  const listOfMothers = findMother(people);

  listOfMothers = onlyWithSon
    ? listOfMothers.filter(child => child.sex === 'm')
    : listOfMothers;

  let sum = 0;

  listOfMothers.forEach((ch) => sum += ch.born - people.find(
    (m) => m.name === ch.mother
  ).born);

  return sum / children.length;
}

function findMother(arrPeople) {
  return ...;
}
```